Simplified/NYPLBookCoverRegistry.m

```diff
@@ -85,6 +85,29 @@ - (NSURL *)URLForPinnedThumbnailImageOfBookIdentifier:(NSString *const)bookIdent
           URLByAppendingPathComponent:[bookIdentifier SHA256]];
 }
 
+- (void)coverImageForBook:(NYPLBook *)book handler:(void (^)(UIImage *image))handler
+{
+
+  //Thumbnail first as placeholder
+  [self thumbnailImageForBook:book handler:handler];
+
+  [[self.session
+    dataTaskWithRequest:[NSURLRequest requestWithURL:book.imageURL]
+    completionHandler:^(NSData *const data,
+                        __attribute__((unused)) NSURLResponse *response,
+                        __attribute__((unused)) NSError *error) {
+      [[NSOperationQueue mainQueue] addOperationWithBlock:^{
```

Comment:
It doesn't seem like this method of fetching a cover image is going to work offline. The app has a notion of "pinned" images that have been saved for offline use, and that notion of pinning is already in use for thumbnails. I would look at `thumbnailImageForBook:handler` to see how you could handle full-size cover images offline as well.